lib/influxer/metrics/metrics.rb

```diff
@@ -162,10 +170,50 @@ def tags
       @attributes.select { |k, _| tag_names.include?(k.to_s) }
     end
 
-    attributes :timestamp
+    def time(parse = false)
+      return unless @time
+
+      return parse_timestamp(@time) if parse
+      @time
+    end
+
+    def time=(val)
+      if val.is_a?(Date) || val.is_a?(Time)
+        @time = parse_time(val)
+      else
+        @time
+      end
+    end
 
     private
 
+    def parse_timestamp(val)
+      case client.time_precision
+      when 'ms'
```

Comment:
Sure, something like that:

```ruby
def parsed_timestamp
      return unless timestamp
      return timestamp unless client.time_precision == 'ms'

      case timestamp
      when String
        (Time.parse(timestamp).to_f * 1000).to_i
      when Date
        (timestamp.to_time.to_f * 1000).to_i
      when Time
        (timestamp.to_f * 1000).to_i
      else
        timestamp
      end
    end
```
Is ignored cases on 'time_precision' is 's' or any value.